packages/ui/src/forum/components/Thread/NewThreadPost.tsx

```diff
@@ -43,57 +43,57 @@ export const NewThreadPost = React.forwardRef(
       setEditable(false)
     }, [])
 
-    if (!active) {
-      return <TextBig ref={ref}>Pick an active membership to post in this thread</TextBig>
-    }
-
```

Comment:
Great suggestion
I think above code should not work, just because we return the showMoal function when we clickn `reply` Button

 const MembershipAndReply = () => {
    showModal<SwitchMemberModalCall>({ modal: 'SwitchMember' })
    replyToPost()
  }

  const onReply = (): void => {
    if (active) return replyToPost()
    return MembershipAndReply()
  }

This could work correctly

Reply:
@thesan Please review the comment by @talentdev0807 :)

Reply:
@talentdev0807 yes you're right this needs to be changed too, although just to keep the logic simpler I would suggest just:
```ts
  const onReply = (): void => {
    if (!active) showModal<SwitchMemberModalCall>({ modal: 'SwitchMember' })
    replyToPost()
  }
```

Reply:
Sounds good